mne/time_frequency/tests/test_psd.py

```diff
@@ -226,3 +226,25 @@ def test_psd_array_welch_n_jobs():
     data = np.zeros((1, 2048))
     psd_array_welch(data, 1024, n_jobs=1)
     psd_array_welch(data, 1024, n_jobs=2)
+
+
+def test_psd_raises_on_inf_in_analyzed_window_array():
+    """psd_array_welch should fail if +Inf lies inside analyzed samples."""
+    n_samples, n_fft, n_overlap = 2048, 256, 128
+    rng = np.random.RandomState(0)
+    x = rng.randn(1, n_samples)
+    # Put +Inf inside the series; this falls within Welch windows
+    x[0, 800] = np.inf
+    with pytest.raises(ValueError, match="non[- ]?finite|NaN|Inf"):
+        psd_array_welch(x, float(n_fft), n_fft=n_fft, n_overlap=n_overlap)
+
+
+def test_psd_raises_on_misaligned_nan_across_channels():
+    """If NaNs are present but masks are NOT aligned across channels, raise."""
+    n_samples, n_fft, n_overlap = 2048, 256, 128
+    rng = np.random.RandomState(42)
+    x = rng.randn(2, n_samples)
+    # NaN only in ch0; ch1 has no NaN => masks not aligned -> should raise
+    x[0, 500] = np.nan
+    with pytest.raises(ValueError, match="aligned|not aligned|non[- ]?finite|NaN|Inf"):
```

Comment:
similar comment as above: avoid complicated regex in the `match` parameter; we wrote the error message so we can pull out a distinctive string of words from it and match on that. That approach makes maintenance easier (you can copy/paste/search on the match string to find the place in the codebase where the error is raised)